modules/monitoring/main.tf

```diff
@@ -12,179 +12,99 @@ resource "azurerm_log_analytics_workspace" "log_analytics_workspace" {
-        "Memory(*)\\% Available Swap Space",
-        "Memory(*)\\Used MBytes Swap Space",
-        "Memory(*)\\% Used Swap Space",
-        "Process(*)\\Pct User Time",
-        "Process(*)\\Pct Privileged Time",
-        "Process(*)\\Used Memory",
-        "Process(*)\\Virtual Shared Memory",
-        "Logical Disk(*)\\% Free Space",
-        "Logical Disk(*)\\% Used Space",
-        "Logical Disk(*)\\Logical Disk Bytes/sec",
-        "Logical Disk(*)\\Disk Read Bytes/sec",
-        "Logical Disk(*)\\Disk Write Bytes/sec",
-        "Logical Disk(*)\\Disk Transfers/sec",
-        "Logical Disk(*)\\Disk Reads/sec",
-        "Logical Disk(*)\\Disk Writes/sec",
-        "Network(*)\\Total Bytes Transmitted",
-        "Network(*)\\Total Bytes Received",
-        "Network(*)\\Total Bytes",
-        "Network(*)\\Total Packets Transmitted",
-        "Network(*)\\Total Packets Received",
-        "Network(*)\\Total Rx Errors",
-        "Network(*)\\Total Tx Errors",
-        "Network(*)\\Total Collisions",
-        "System(*)\\Uptime",
-        "System(*)\\Users",
-        "System(*)\\Unique Users",
-        "System(*)\\CPUs"
-      ]
-      name = local.performance_counter_name
-    }
+resource "azurerm_application_insights_smart_detection_rule" "trace_severity_ratio_degradation" {
+  name                    = "Degradation in trace severity ratio"
+  application_insights_id = azurerm_application_insights.graphdb_insights.id
+  enabled                 = false
+}
+
+# Availability tests
+resource "azurerm_application_insights_standard_web_test" "example" {
+  enabled                          = true
+  name                    = "cluster-health2"
+  resource_group_name     = var.resource_group_name
+  location                = var.location
+  application_insights_id = azurerm_application_insights.graphdb_insights.id
+  geo_locations           = ["us-va-ash-azr"]
+
+  request {
+    url = "https://${var.availability_request_url}/rest/cluster/node/status"
+    parse_dependent_requests_enabled = false
   }
 
-  stream_declaration {
-    stream_name = "Custom-${azapi_resource.main_gdb_log_table.name}"
-    column {
-      name = "TimeGenerated"
-      type = "datetime"
-    }
-    column {
-      name = "RawData"
-      type = "string"
+  validation_rules {
+    expected_status_code             = 200
+    ssl_check_enabled                = false
+    content {
+      content_match      = "\"nodeState\":\"LEADER\""
+      pass_if_text_found = true
     }
   }
-}
 
-resource "azurerm_monitor_data_collection_rule_association" "DCR_association" {
-  name                    = "dcra-${var.resource_group_name}"
-  target_resource_id      = var.vmss_resource_id
-  data_collection_rule_id = azurerm_monitor_data_collection_rule.data_collection_rule.id
-  description             = "Associates DCR to the VMSS"
-}
 
-# We need to use the azapi as terraform does not have a resource to create tables.
-resource "azapi_resource" "main_gdb_log_table" {
```

Comment:
You no longer need the provider for azapi

Reply:
removed